tests/client_request.rs

```diff
@@ -440,9 +440,14 @@ fn request_with_connection_headers() {
     let _ = ::env_logger::try_init();
     let (io, srv) = mock::new();
 
-    let srv = srv.assert_client_handshake()
+    // can't assert full handshake, since client never send a request, and
+    // thus never boths to ack the settings...
+    let srv = srv.read_preface()
         .unwrap()
-        .recv_settings()
+        .recv_frame(frames::settings())
+        // goaway is just to make sure the connection closes because
+        // of no active streams, not an error
+        .recv_frame(frames::go_away(0))
```

Comment:
Would this test fail without the change? Even with the comments, it's not immediately clear to me how this test exercises the error case.

Reply:
Yep, by adding this goaway receive check, the test will hang, even though all sender handles have been dropped, because the connection thinks there's some 6 active streams.